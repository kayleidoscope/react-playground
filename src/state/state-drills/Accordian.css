/* Yes I know I spelled accordion wrong */
h1 {
    text-align: center;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: xx-large;
}

ul {
    padding: 0;
}

li {
    list-style: none;
    background-color: lightsteelblue;
    padding-bottom: 15px;
}

button {
    border-bottom: groove;
    background-color: lightgray;
    width: 100%;
    padding: 5px;
    font-size: x-large;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.content {
    margin: 0;
    padding: 15px;
}