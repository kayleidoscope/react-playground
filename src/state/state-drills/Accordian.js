import React from "react";
import './Accordian.css'

//Yes I know I spelled accordion wrong

class Accordian extends React.Component {
    static defaultProps = {
        sections: []
    };

    state = {
        currentTabIndex: null
    }

    handleButtonClick(index) {
        console.log('button clicked!', { index })
        this.setState({currentTabIndex: index })
    }

    renderButtons() {
        return this.props.sections.map((section, index) => (
            <li 
                key={index}>
                <button 
                    className="accordion-button"
                    onClick={() => this.handleButtonClick(index)}
                >
                    {section.title}
                </button>
                {this.renderContent(section, index)}
            </li>
        ))
    }
    renderContent(section, index) {
        if (this.state.currentTabIndex === index) {
            return (
                <p className='content'>
                    {section.content}
                </p>
            )
        }
    }
  render() {
    return (
        <div>
            <ul>
                {this.renderButtons()}
            </ul>
        </div>
    )}
}

export default Accordian;