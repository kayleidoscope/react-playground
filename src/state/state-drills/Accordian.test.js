import React from 'react';
import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';

//Yes I know I spelled accordion wrong

describe(`Accordian Component`, () => {
    // array of objects, each with a name and content
    const sections = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ]
    
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders no section by default', () => {
        const wrapper = shallow(<Accordian sections={sections} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders empty given no tabs', () => {
        const wrapper = shallow(<Accordian />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('closes the first tab and opens any clicked tab', () => {
        const wrapper = shallow(<Accordian sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})