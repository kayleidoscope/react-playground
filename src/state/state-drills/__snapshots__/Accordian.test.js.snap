// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component closes the first tab and opens any clicked tab 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p
        className="content"
      >
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordian Component renders empty given no tabs 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordian Component renders no section by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
